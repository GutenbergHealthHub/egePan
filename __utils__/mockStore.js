
// (C) Copyright IBM Deutschland GmbH 2021.  All rights reserved.

/***********************************************************************************************
imports
***********************************************************************************************/

// redux
/*-----------------------------------------------------------------------------------*/

import thunk from 'redux-thunk'
import { combineReducers, createStore, applyMiddleware } from 'redux'

// reducers
/*-----------------------------------------------------------------------------------*/

import LoginReducer from '../src/screens/login/loginReducer'
import AboutReducer from '../src/screens/about/aboutReducer'
import CheckInReducer from '../src/screens/checkIn/checkInReducer'

// services
/*-----------------------------------------------------------------------------------*/

import localStorage from '../src/services/localStorage/localStorage'

/***********************************************************************************************
reducer
***********************************************************************************************/

const appReducer = combineReducers({
	Login: LoginReducer,
	CheckIn: CheckInReducer,
	About: AboutReducer
})

const rootReducer = (state, action) => {

	// global actions that need to be available anywhere,
	// will be filtered out and executed by the rootReducer
	// (before the actual reducer is)

	if (action.type === 'USER_LOGOUT') {
		state = undefined
		localStorage.removeLastSubjectId()
	}

	if (action.type === 'DELETE_ALL_LOCAL_DATA') {
		localStorage.clearAll()
	}

	if (action.type === 'DELETE_LOCAL_QUESTIONNAIRE') {
		state.CheckIn.currentCategoryIndex = null
		state.CheckIn.questionnaireItemMap = null
		state.CheckIn.categories = null
		state.CheckIn.categoriesLoaded = false
		state.CheckIn.currentPageIndex = 1

		let subjectId = state.CheckIn.user.subjectId
		localStorage.removeLastQuestionnaireId(subjectId)
		localStorage.removeCategories(subjectId)
		localStorage.removeQuestionnaireItemMap(subjectId)
	}

	return appReducer(state, action)
}

/***********************************************************************************************
middleware
***********************************************************************************************/

const middleware = []
middleware.push(thunk)

// if(__DEV__) middleware.push(createLogger({collapsed: true}))

/***********************************************************************************************
export
***********************************************************************************************/

let globalMockStore

export function buildStore(state) {
	if(state) globalMockStore = createStore(rootReducer, state, applyMiddleware(...middleware))
	if(!state) globalMockStore = createStore(rootReducer, applyMiddleware(...middleware))
	
	return globalMockStore
}

export default buildStore()