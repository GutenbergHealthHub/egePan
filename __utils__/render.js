import React from 'react'
import { Provider } from 'react-redux'
import { render } from '@testing-library/react-native'
import { buildStore } from './mockStore'

let store = null
 
export function renderWithRedux(subComponent, state) {

    if(!store) {
        state = state || {}
        store = buildStore(state)
    }

    const queries = render(<Provider store={store}>{subComponent}</Provider>)
    
    return {
        ...queries,
        store,
    }
}

export function initRenderModule(initialState) {

    initialState = initialState || {}
    
    store = buildStore(initialState)

    return store
}