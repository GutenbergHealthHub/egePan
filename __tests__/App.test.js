// (C) Copyright IBM Deutschland GmbH 2021.  All rights reserved.

/***********************************************************************************************
imports
***********************************************************************************************/

import React from 'react'
import { expect } from '@jest/globals'
import { renderWithRedux } from '../__utils__/render'
import createAppNavigator from '../src/navigation/appNavigator'

import App from '../App'

/***********************************************************************************************
tests
***********************************************************************************************/

describe('APP START AND RENDERING:', () => {

    let createAppTree = () => {
    
        let Navigator = createAppNavigator()
            
        return renderWithRedux(<Navigator><App /></Navigator>);   
    }

    // simple render test
    test ('<App /> can be created and rendered', () => {

         // renders the component
        let tree = createAppTree()

        // checks if the screen matches the snapshot
        expect(tree).toMatchSnapshot()
    })

    // checks if App produces exactly one child
    test ('<App /> has exactly one child', () => {
        
        expect(createAppTree().toJSON().children.length).toBe(1)
    })

    // after the app start, the user should be automatically routed to the landing screen
    test ('User is navigated to <LandingScreen /> after app start', () => {
    
        // checks if right now the login-button is rendered
        let loginButton = createAppTree().getByText('Navigate to Login Screen')
        
        expect(loginButton).toBeTruthy()
    })
})
