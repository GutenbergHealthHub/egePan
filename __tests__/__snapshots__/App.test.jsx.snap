// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APP START AND RENDERING: <App /> can be created and rendered 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          Object {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            collapsable={false}
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "marginTop": 0,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              handlerTag={1}
              handlerType="PanGestureHandler"
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#FFFFFF",
                          "flexDirection": "column",
                          "height": "100%",
                        }
                      }
                    >
                      <Modal
                        hardwareAccelerated={false}
                        visible={false}
                      />
                      <View
                        style={
                          Object {
                            "alignContent": "stretch",
                            "alignItems": "stretch",
                            "backgroundColor": "#FFFFFF",
                            "flex": 1,
                            "flexDirection": "column",
                            "justifyContent": "flex-start",
                            "paddingTop": 30,
                            "width": "100%",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "flexDirection": "row",
                              "flexWrap": "wrap",
                              "justifyContent": "space-between",
                              "paddingTop": 10,
                              "width": "100%",
                              "zIndex": 2,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "alignItems": "center",
                                "flexDirection": "row",
                                "flexWrap": "wrap",
                                "justifyContent": "flex-end",
                                "marginLeft": 15,
                                "marginRight": 15,
                                "textAlign": "center",
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "height": 44,
                                  "width": 104.65116279069767,
                                }
                              }
                            />
                          </View>
                        </View>
                        <Image
                          resizeMode="contain"
                          source={
                            Object {
                              "testUri": "../../../src/CUSTOMIZATION/images/logo.png",
                            }
                          }
                          style={
                            Object {
                              "alignSelf": "center",
                              "flex": 1,
                              "marginVertical": 5,
                              "position": "relative",
                            }
                          }
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "backgroundColor": "#FFFFFF",
                            "flex": 2,
                            "flexDirection": "column",
                            "height": "100%",
                          }
                        }
                      >
                        <View
                          onLayout={[Function]}
                          style={
                            Object {
                              "flex": 1,
                              "flexDirection": "column",
                              "width": "100%",
                            }
                          }
                        >
                          <RCTScrollView
                            contentContainerStyle={
                              Object {
                                "flexGrow": 1,
                              }
                            }
                            contentData={
                              <View
                                style={
                                  Object {
                                    "backgroundColor": "#FFFFFF",
                                    "flexDirection": "column",
                                    "height": "100%",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "justifyContent": "center",
                                      "marginBottom": 61.04651162790697,
                                      "paddingTop": 61.04651162790697,
                                    }
                                  }
                                >
                                  <Themed.Text
                                    style={
                                      Object {
                                        "alignSelf": "center",
                                        "fontFamily": "IBMPlexSans-SemiBold",
                                        "fontSize": 33.75,
                                        "lineHeight": 43.125,
                                        "textAlign": "center",
                                        "width": "80%",
                                      }
                                    }
                                  >
                                    Willkommen in der S.A.M.-Studien-App
                                  </Themed.Text>
                                  <Themed.Text
                                    style={
                                      Object {
                                        "alignSelf": "center",
                                        "color": "#1F2532",
                                        "fontFamily": "IBMPlexSans",
                                        "fontSize": 28.125,
                                        "lineHeight": 37.5,
                                        "marginBottom": 34.883720930232556,
                                        "marginLeft": 69.76744186046511,
                                        "marginRight": 69.76744186046511,
                                        "marginTop": 34.883720930232556,
                                        "textAlign": "center",
                                      }
                                    }
                                  >
                                    Klicken Sie auf den "Registrieren"-Button, um sich neu zu registrieren.

 Falls sie bereits registriert sind, geben Sie bitte Ihre Probanden-ID ein, um sich wieder anzumelden.
                                  </Themed.Text>
                                </View>
                                <View
                                  style={
                                    Object {
                                      "flex": 1,
                                      "height": "100%",
                                      "justifyContent": "flex-end",
                                      "marginBottom": 36,
                                      "marginTop": 20,
                                    }
                                  }
                                >
                                  <TouchableOpacity
                                    accessibilityHint="Startet den Login-Prozess"
                                    accessibilityLabel="Registrieren"
                                    accessibilityRole="button"
                                    onPress={[Function]}
                                    style={
                                      Object {
                                        "alignSelf": "center",
                                        "backgroundColor": "#376991",
                                        "borderColor": "#376991",
                                        "borderRadius": 50,
                                        "borderWidth": 0,
                                        "bottom": 0,
                                        "padding": 15,
                                        "paddingBottom": 26.162790697674417,
                                        "paddingLeft": 26.162790697674417,
                                        "paddingRight": 26.162790697674417,
                                        "paddingTop": 26.162790697674417,
                                        "width": "80%",
                                      }
                                    }
                                    testID="registerButton"
                                  >
                                    <Themed.Text
                                      style={
                                        Object {
                                          "alignSelf": "center",
                                          "color": "#FFFFFF",
                                          "textAlign": "center",
                                        }
                                      }
                                    >
                                      Registrieren
                                    </Themed.Text>
                                  </TouchableOpacity>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "alignSelf": "center",
                                        "borderColor": "#a5353e",
                                        "borderRadius": 50,
                                        "borderWidth": 1,
                                        "flexDirection": "row",
                                        "marginTop": 10,
                                        "paddingHorizontal": 10,
                                        "width": "80%",
                                      }
                                    }
                                  >
                                    <ForwardRef(Themed.Input)
                                      accessibilityHint="Bitte geben Sie Ihre Probanden-ID ein, sofern sie bereits registriert sind"
                                      accessibilityLabel="Probanden-ID eingeben"
                                      autoCapitalize="none"
                                      autoCorrect={false}
                                      containerStyle={
                                        Object {
                                          "width": "80%",
                                        }
                                      }
                                      errorMessage={null}
                                      inputContainerStyle={
                                        Object {
                                          "borderBottomWidth": 0,
                                        }
                                      }
                                      inputStyle={
                                        Object {
                                          "fontSize": undefined,
                                        }
                                      }
                                      onChangeText={[Function]}
                                      placeholder="Probanden-ID eingeben"
                                      renderErrorMessage={false}
                                      value={null}
                                    />
                                    <Themed.Button
                                      accessibilityHint="Abschicken"
                                      accessibilityLabel="Login"
                                      accessibilityRole="button"
                                      onPress={[Function]}
                                      title="Login"
                                      titleStyle={
                                        Object {
                                          "fontSize": undefined,
                                        }
                                      }
                                      type="clear"
                                    />
                                  </View>
                                </View>
                              </View>
                            }
                            enableOnAndroid={true}
                            innerRef={[Function]}
                            onScroll={[Function]}
                            style={
                              Object {
                                "flex": 1,
                                "flexDirection": "column",
                                "width": "100%",
                              }
                            }
                          >
                            <View>
                              <View
                                style={
                                  Object {
                                    "backgroundColor": "#FFFFFF",
                                    "flexDirection": "column",
                                    "height": "100%",
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "justifyContent": "center",
                                      "marginBottom": 61.04651162790697,
                                      "paddingTop": 61.04651162790697,
                                    }
                                  }
                                >
                                  <Text
                                    style={
                                      Object {
                                        "alignSelf": "center",
                                        "color": "#242424",
                                        "fontFamily": "IBMPlexSans-SemiBold",
                                        "fontSize": 33.75,
                                        "lineHeight": 43.125,
                                        "textAlign": "center",
                                        "width": "80%",
                                      }
                                    }
                                  >
                                    Willkommen in der S.A.M.-Studien-App
                                  </Text>
                                  <Text
                                    style={
                                      Object {
                                        "alignSelf": "center",
                                        "color": "#1F2532",
                                        "fontFamily": "IBMPlexSans",
                                        "fontSize": 28.125,
                                        "lineHeight": 37.5,
                                        "marginBottom": 34.883720930232556,
                                        "marginLeft": 69.76744186046511,
                                        "marginRight": 69.76744186046511,
                                        "marginTop": 34.883720930232556,
                                        "textAlign": "center",
                                      }
                                    }
                                  >
                                    Klicken Sie auf den "Registrieren"-Button, um sich neu zu registrieren.

 Falls sie bereits registriert sind, geben Sie bitte Ihre Probanden-ID ein, um sich wieder anzumelden.
                                  </Text>
                                </View>
                                <View
                                  style={
                                    Object {
                                      "flex": 1,
                                      "height": "100%",
                                      "justifyContent": "flex-end",
                                      "marginBottom": 36,
                                      "marginTop": 20,
                                    }
                                  }
                                >
                                  <View
                                    accessibilityHint="Startet den Login-Prozess"
                                    accessibilityLabel="Registrieren"
                                    accessibilityRole="button"
                                    accessible={true}
                                    collapsable={false}
                                    focusable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "alignSelf": "center",
                                        "backgroundColor": "#376991",
                                        "borderColor": "#376991",
                                        "borderRadius": 50,
                                        "borderWidth": 0,
                                        "bottom": 0,
                                        "opacity": 1,
                                        "padding": 15,
                                        "paddingBottom": 26.162790697674417,
                                        "paddingLeft": 26.162790697674417,
                                        "paddingRight": 26.162790697674417,
                                        "paddingTop": 26.162790697674417,
                                        "width": "80%",
                                      }
                                    }
                                    testID="registerButton"
                                  >
                                    <Text
                                      style={
                                        Object {
                                          "alignSelf": "center",
                                          "color": "#FFFFFF",
                                          "textAlign": "center",
                                        }
                                      }
                                    >
                                      Registrieren
                                    </Text>
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "alignSelf": "center",
                                        "borderColor": "#a5353e",
                                        "borderRadius": 50,
                                        "borderWidth": 1,
                                        "flexDirection": "row",
                                        "marginTop": 10,
                                        "paddingHorizontal": 10,
                                        "width": "80%",
                                      }
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "paddingHorizontal": 10,
                                          "width": "80%",
                                        }
                                      }
                                    >
                                      <View
                                        collapsable={false}
                                        style={
                                          Object {
                                            "alignItems": "center",
                                            "borderBottomWidth": 0,
                                            "borderColor": "#86939e",
                                            "flexDirection": "row",
                                            "transform": Array [
                                              Object {
                                                "translateX": 0,
                                              },
                                            ],
                                          }
                                        }
                                      >
                                        <TextInput
                                          accessibilityHint="Bitte geben Sie Ihre Probanden-ID ein, sofern sie bereits registriert sind"
                                          accessibilityLabel="Probanden-ID eingeben"
                                          autoCapitalize="none"
                                          autoCorrect={false}
                                          editable={true}
                                          onChangeText={[Function]}
                                          placeholder="Probanden-ID eingeben"
                                          placeholderTextColor="#86939e"
                                          style={
                                            Object {
                                              "color": "#242424",
                                              "flex": 1,
                                              "fontSize": undefined,
                                              "minHeight": 40,
                                            }
                                          }
                                          testID="RNE__Input__text-input"
                                          underlineColorAndroid="transparent"
                                          value={null}
                                        />
                                      </View>
                                      <Text
                                        style={
                                          Object {
                                            "color": "#ff190c",
                                            "fontSize": 12,
                                            "height": 0,
                                            "margin": 0,
                                            "padding": 0,
                                          }
                                        }
                                      />
                                    </View>
                                    <View
                                      style={
                                        Array [
                                          Object {
                                            "borderRadius": 3,
                                            "overflow": "hidden",
                                          },
                                          Object {
                                            "borderRadius": 3,
                                          },
                                          undefined,
                                          false,
                                        ]
                                      }
                                    >
                                      <View
                                        accessibilityHint="Abschicken"
                                        accessibilityLabel="Login"
                                        accessibilityRole="button"
                                        accessibilityState={
                                          Object {
                                            "busy": false,
                                            "disabled": false,
                                          }
                                        }
                                        accessible={true}
                                        collapsable={false}
                                        focusable={true}
                                        onClick={[Function]}
                                        onResponderGrant={[Function]}
                                        onResponderMove={[Function]}
                                        onResponderRelease={[Function]}
                                        onResponderTerminate={[Function]}
                                        onResponderTerminationRequest={[Function]}
                                        onStartShouldSetResponder={[Function]}
                                        style={
                                          Object {
                                            "opacity": 1,
                                          }
                                        }
                                      >
                                        <View
                                          style={
                                            Object {
                                              "alignItems": "center",
                                              "backgroundColor": "transparent",
                                              "borderColor": "#2089dc",
                                              "borderRadius": 3,
                                              "borderWidth": 0,
                                              "flexDirection": "row",
                                              "justifyContent": "center",
                                              "padding": 8,
                                            }
                                          }
                                        >
                                          <Text
                                            style={
                                              Object {
                                                "color": "#2089dc",
                                                "fontSize": undefined,
                                                "paddingVertical": 1,
                                                "textAlign": "center",
                                              }
                                            }
                                          >
                                            Login
                                          </Text>
                                        </View>
                                      </View>
                                    </View>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </RCTScrollView>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
    <View
      collapsable={false}
      pointerEvents="box-none"
      style={
        Object {
          "height": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        }
      }
    />
  </RNCSafeAreaProvider>
</View>
`;
